parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueReaders.java

```diff
@@ -401,6 +404,17 @@ public ByteBuffer read(ByteBuffer reuse) {
     }
   }
 
+  public static class UUIDReader extends PrimitiveReader<UUID> {
```

Comment:
Rather than making the classes here public, I would prefer to introduce factory methods like we have for the Avro readers. I think that gives us more flexibility to change implementations and move classes around without worrying because none of them need to be public.